### ISSUE TITLE
guidance on mAP improvement

### ISSUE THREAD

**Issue by edeyejedi** [ultralytics/yolov5#590]
## ❔Question
I have run training on a data set of ~2000 images, split 70/20/10 on train/val/test. I am managing to get peak mAP0.5 values of ~0.6, which I think is pretty good, in fact it is really good when used on the test data....but i would like it to be better. 
See attached plot of training results. The mAP0.5 plateaus really early in all of the training runs. I have run all the different size models - following the prescribed batch sizes; used pretrained weights and tried from scratch; tried one with an Adam optimizer with reduced LR0 (v.little difference); and played a lot with how I define the object - size of box. The jump from mAP0.5 of ~0.3 to ~0.6 was using slightly larger boxes - some objects are really small relative to the image size
Obviously I could expand my training set. But I was wondering if the following would be worth exploring, or a complete waste of time:
- train on slightly higher resolution/size images (currently 416)
- using grey scale, would RGB be better
- switch off the augmentation?? or parts of it?? The idea of using mirror images for this particular application, where objects are always orientated a certain way and the field of view of the camera is always the same seems, intuitively, counter productive. I could be extremely wrong there............
-Dropping the learning rate right down, have seen better results across a bunch of platforms from this approach
- Dive in to the hyper-parameters (beyond optimizer and Lr0 ), but where to start??

I admit that I have a fair bit of reading to do get a handle on the role of all the hyperparamters, but would be great if anyone has some pointers. Thanks for even reading this far! look forward to some input. 

![Results](https://user-images.githubusercontent.com/64930337/89090230-b415d780-d3f5-11ea-82cc-5b346927f6f5.png)
![results](https://user-images.githubusercontent.com/64930337/89090244-c728a780-d3f5-11ea-9a65-6d1c8ef4d907.png)
![labels](https://user-images.githubusercontent.com/64930337/89090257-eaebed80-d3f5-11ea-89d9-632051a52406.png)

**Comment by glenn-jocher** [comment]
@edeyejedi I would recommend starting from default settings.

**Comment by edeyejedi** [comment]
Hi @glenn-jocher 
yep, already did that.

**Comment by EricWiener** [comment]
@edeyejedi How were you able to graph the performance of the model?

**Comment by edeyejedi** [comment]
The top plot i just whipped up in python, the other plots are directly from the runs folder. 
FYI - i significantly improved this model by increasing the size of the images used for training.

**Comment by EricWiener** [comment]
Thanks for the response and the tip @edeyejedi ! Were you able to evaluate the final model on just test data and get recall, precision, etc. without having to re-train the model?

**Comment by glenn-jocher** [comment]
@edeyejedi you can set an augmentation policy here, including disabling left-right flips:
https://github.com/ultralytics/yolov5/blob/d2da5230533db7a2c76af1dde6d91c7e1631a1b8/data/hyp.finetune.yaml#L1-L27

You can try Hyperparameter Evolution as well, if you have resources and time:
https://docs.ultralytics.com/yolov5


In general there is no free lunch. To improve your results:
- Use a larger model
- Gather more data
- Train and test at a higher resolution
- Evolve your hyperparameters

**Comment by edeyejedi** [comment]
@EricWiener not sure what you mean. the model performs well. 
@glenn-jocher yep, been doing that one for a while. Have been systematically changing hyper parameters and plotting results. Have not manage to run an evolution yet (well I did when i first picked up Yv5 but didn't really know what I was doing), but will look at doing that when i can secure something a little more reliable than colab.
